### PR TITLE
FEATURE: Cook drafts excerpt in user activity

### PR DESCRIPTION
**Pull request by udan11**
The previous excerpt was a simple truncated raw message. Starting with
this commit, the raw content of the draft is cooked and an excerpt is
extracted from it. The logic for extracting the excerpt mimics the the
`ExcerptParser` class, but does not implement all functionality, being
a much simpler implementation.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/text.js [hunk 1]
```diff
@@ -110,3 +110,55 @@ export function emojiUrlFor(code) {
...
+function encode(str) {
+  return str.replaceAll("<", "&lt;").replaceAll(">", "&gt;");
+}
+
+function traverse(element, callback) {
+  if (callback(element)) {
+    for (let i = 0; i < element.childNodes.length; ++i) {
+      traverse(element.childNodes[i], callback);
+    }
+  }
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/app/lib/text.js:123]: Any reasons not to use ` forEach`  instead?

` ` ` 
element.childNodes.forEach(child => traverse(child, callback))
` ` `

File: config/routes.rb [hunk 2]
```diff
@@ -887,10 +887,10 @@ def patch(*) end # Disable PATCH requests
 
     get "message-bus/poll" => "message_bus#poll"
 
-    resources :drafts, only: [:index]
-    get "draft" => "draft#show"
-    post "draft" => "draft#update"
-    delete "draft" => "draft#destroy"
+    get "drafts" => "drafts#index"
+    get "draft" => "drafts#show"
```
Comment (ZogStriP) [inline config/routes.rb:891]: Hmm, those routes should probably all be plurals?

**Review by ZogStriP — APPROVED** [review]
LGTM 👍

**Comment by udan11** [comment]
The plugins frontend test failure is legit. I have a PR ready for it discourse/discourse-encrypt#135.

**Pull request merged**